app/models/sample.rb

```diff
@@ -387,6 +387,21 @@ def self.editable(user)
     end
   end
 
+  def deletable?(user)
+    if user.admin?
+      return true
+    elsif user_id == user.id
+      # Sample belongs to the user
+      if pipeline_runs.empty? ||
+         (pipeline_runs[0].job_status.include?("Filtering-FAILED") || pipeline_runs[0].job_status.include?("alignment-FAILED"))
```

Comment:
is there a way to avoid hard-coding this? How about checking the status of the output_states

Reply:
You can use REPORT_READY_OUTPUT to check whether the failure should allow the sample to be deletable or not